### PR TITLE
simulationmachine

### PR DESCRIPTION
**Pull request by kevinykuo**
Ready for review. Do not merge before https://github.com/kasaai/simulationmachine/issues/20 and https://github.com/kasaai/simulationmachine/issues/21 are resolved.

### CODE REVIEW

**Comment by hdykiel** [comment]
Great post. The one thing I'd consider changing is the sentence "If you're a practitioner or researcher looking for realistic data for testing reserving or claims forecasting models, you may find it interesting". It could be changed to something like "We created this package for the practitioner or researcher who needs realistic data for testing reserving or claims forecasting models."

**Comment by 12tafran** [comment]
Do you intend for the `simulation_machine()` to only use the Log-normal distribution to simulate loss and recovery? If you are, that is perfectly fine.  If you are planning to add additional distribution, maybe add something in the motivation and technical detail section that mention Log-normal isn't the only distribution and additional loss distribution will be made available for wider use

**Comment by kevinykuo** [comment]
@12tafran yep the model is limited to lognormal claim sizes

**Comment by kevinykuo** [comment]
@actuarialvoodoo could you take one last look before we ship?

**Comment by ryanbthomas** [comment]
@kevinykuo I will review it today and provide any comments this evening.

File: _posts/simulation-machine/individual-claims-history-simulation-machine.Rmd [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+## simulationmachine
+
+You can download simulationmachine from GitHub with
+
+```r
+# install.packages("remotes")
+remotes::install_github("kasaai/simulationmachine")
+```
+
+First, specify the desired characteristics of the simulation:
```
Comment (ryanbthomas) [inline _posts/simulation-machine/individual-claims-history-simulation-machine.Rmd]: I think a short paragraph and link to the conjure package would increase clarity here because I think readers might get hung up on the variable "charm". It makes a lot more sense if one understands the metaphor used in the conjure package.

File: _posts/simulation-machine/individual-claims-history-simulation-machine.Rmd [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  num_claims = 50000, 
+  lob_distribution = c(0.25, 0.25, 0.30, 0.20), 
+  inflation = c(0.01, 0.01, 0.01, 0.01), 
+  sd_claim = 0.85, 
+  sd_recovery = 0.75
+)
+charm
+```
+
+Here, we are noting that we would like
```
Comment (ryanbthomas) [inline _posts/simulation-machine/individual-claims-history-simulation-machine.Rmd]: "The charm encodes the details of the simulation we want to create. In this case, the simulation contains the following"

File: _posts/simulation-machine/individual-claims-history-simulation-machine.Rmd [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+- Inflation is a constant 1% for all lines of business
+- Claims are lognormally distributed with a standard deivation of 0.85
+- Recoveries are lognormally distributed with a standard deviation of 0.75
+
+Once we have the specification, we can perform the actual simulation using the `conjure()` function:
+
+```{r, echo = TRUE, message = FALSE}
+library(tidyverse)
+records <- conjure(charm, seed = 100)
+glimpse(records)
```
Comment (ryanbthomas) [inline _posts/simulation-machine/individual-claims-history-simulation-machine.Rmd:71]: I stared at 600k for a min before I guessed it was 50k claims x 12 periods of obs. I know this is highlighted at the bottom, but I think more context on the details from the paper would be helpful. Before I got to the end of the paper, I didn't realize that the `simulationmachine` was tied to a specific dataset. This makes the parameters and some of the outputs (e.g., lognormal distribution and AY 1994 - 2005) make more sense. Perhaps, adding a brief summary of the paper (a condensed version of 2nd paragraph in 'Motivations & Technical Details' section) to the end of the introduction section would help to "set the stage" for readers.

Separately from that, I think a brief comment of why 600k records here is worthwhile.

File: _site/posts/simulation-machine/bibliography.bib [hunk 4]
```diff
@@ -0,0 +1,10 @@
+@article{gabrielli2018individual,
```
Comment (ryanbthomas) [inline _site/posts/simulation-machine/bibliography.bib:1]: Can we add a link so that readers can get to the original paper easily?

File: _posts/simulation-machine/individual-claims-history-simulation-machine.Rmd [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  # calculate cumulative losses
+  mutate(cumulative_paid_loss = cumsum(paid_loss)) %>% 
+  select(accident_year, development_year, cumulative_paid_loss) %>% 
+  # reshape the data
+  pivot_wider(
+    names_from = "development_year", 
+    values_from = "cumulative_paid_loss"
+  )
+```
+
```
Comment (ryanbthomas) [inline _posts/simulation-machine/individual-claims-history-simulation-machine.Rmd]: Remove blank line
